Integration tests fail when run using Python 3

**Summary**
The integration tests (everything under `tests/`) fail when run using Python 3.

**Steps To Reproduce (STR)**
1. Build a Python 3 `virtualenv`, `pip` install `requirements/test.txt`, and activate it
2. Run `pytest --base-url https://developer.allizom.org tests`

**Actual behavior**
You'll see this error:
```
=================================================== ERRORS ===================================================
_____________________________ ERROR collecting tests/headless/test_endpoints.py ______________________________
../../virtualenvs/kuma-test/lib/python3.7/site-packages/_pytest/runner.py:157: in __init__
    self.result = func()
../../virtualenvs/kuma-test/lib/python3.7/site-packages/_pytest/runner.py:328: in <lambda>
    lambda: list(collector.collect()),
../../virtualenvs/kuma-test/lib/python3.7/site-packages/_pytest/python.py:402: in collect
    return super(Module, self).collect()
../../virtualenvs/kuma-test/lib/python3.7/site-packages/_pytest/python.py:314: in collect
    res = self.makeitem(name, obj)
../../virtualenvs/kuma-test/lib/python3.7/site-packages/_pytest/python.py:326: in makeitem
    collector=self, name=name, obj=obj)
../../virtualenvs/kuma-test/lib/python3.7/site-packages/_pytest/vendored_packages/pluggy.py:745: in __call__
    return self._hookexec(self, self._nonwrappers + self._wrappers, kwargs)
../../virtualenvs/kuma-test/lib/python3.7/site-packages/_pytest/vendored_packages/pluggy.py:339: in _hookexec
    return self._inner_hookexec(hook, methods, kwargs)
../../virtualenvs/kuma-test/lib/python3.7/site-packages/_pytest/vendored_packages/pluggy.py:334: in <lambda>
    _MultiCall(methods, kwargs, hook.spec_opts).execute()
../../virtualenvs/kuma-test/lib/python3.7/site-packages/_pytest/vendored_packages/pluggy.py:613: in execute
    return _wrapped_call(hook_impl.function(*args), self.execute)
../../virtualenvs/kuma-test/lib/python3.7/site-packages/_pytest/vendored_packages/pluggy.py:250: in _wrapped_call
    wrap_controller.send(call_outcome)
../../virtualenvs/kuma-test/lib/python3.7/site-packages/_pytest/python.py:186: in pytest_pycollect_makeitem
    res = list(collector._genfunctions(name, obj))
../../virtualenvs/kuma-test/lib/python3.7/site-packages/_pytest/python.py:346: in _genfunctions
    self.ihook.pytest_generate_tests(metafunc=metafunc)
../../virtualenvs/kuma-test/lib/python3.7/site-packages/_pytest/vendored_packages/pluggy.py:745: in __call__
    return self._hookexec(self, self._nonwrappers + self._wrappers, kwargs)
../../virtualenvs/kuma-test/lib/python3.7/site-packages/_pytest/vendored_packages/pluggy.py:339: in _hookexec
    return self._inner_hookexec(hook, methods, kwargs)
../../virtualenvs/kuma-test/lib/python3.7/site-packages/_pytest/vendored_packages/pluggy.py:334: in <lambda>
    _MultiCall(methods, kwargs, hook.spec_opts).execute()
../../virtualenvs/kuma-test/lib/python3.7/site-packages/_pytest/vendored_packages/pluggy.py:614: in execute
    res = hook_impl.function(*args)
../../virtualenvs/kuma-test/lib/python3.7/site-packages/_pytest/python.py:113: in pytest_generate_tests
    metafunc.parametrize(*marker.args, **marker.kwargs)
../../virtualenvs/kuma-test/lib/python3.7/site-packages/_pytest/python.py:841: in parametrize
    ids = idmaker(argnames, parameters, idfn, ids, self.config)
../../virtualenvs/kuma-test/lib/python3.7/site-packages/_pytest/python.py:966: in idmaker
    for valindex, parameterset in enumerate(parametersets)]
../../virtualenvs/kuma-test/lib/python3.7/site-packages/_pytest/python.py:966: in <listcomp>
    for valindex, parameterset in enumerate(parametersets)]
../../virtualenvs/kuma-test/lib/python3.7/site-packages/_pytest/python.py:956: in _idvalset
    if ids is None or (idx >= len(ids) or ids[idx] is None):
E   TypeError: 'dict_keys' object is not subscriptable
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!! Interrupted: 1 errors during collection !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
========================================== 1 error in 0.52 seconds ===========================================
```

**Expected behavior**
The tests to run successfully.

--- a/tests/conftest.py
+++ b/tests/conftest.py
@@ -84,7 +84,7 @@ def is_local_url(base_url):
     instance (localhost).
     """
     return (base_url and
-            urlsplit(base_url).hostname.split('.')[-1] == 'localhost')
+            'localhost' in urlsplit(base_url).hostname.split('.'))
 
 
 @pytest.fixture(scope='session')

--- a/tests/headless/test_endpoints.py
+++ b/tests/headless/test_endpoints.py
@@ -111,7 +111,7 @@ def test_document(site_url, is_indexed):
     resp = request('get', url)
     assert resp.status_code == 200
     assert resp.headers['Content-Type'] == 'text/html; charset=utf-8'
-    meta = META_ROBOTS_RE.search(resp.content)
+    meta = META_ROBOTS_RE.search(resp.text)
     assert meta
     content = meta.group('content')
     if is_indexed:
@@ -127,7 +127,7 @@ def test_user_document(site_url):
     resp = request('get', url)
     assert resp.status_code == 200
     assert resp.headers['Content-Type'] == 'text/html; charset=utf-8'
-    meta = META_ROBOTS_RE.search(resp.content)
+    meta = META_ROBOTS_RE.search(resp.text)
     assert meta
     content = meta.group('content')
     # Pages with legacy MindTouch namespaces like 'User:' never get
@@ -168,7 +168,7 @@ def test_home(site_url, is_indexed):
     resp = request('get', url)
     assert resp.status_code == 200
     assert resp.headers['Content-Type'] == 'text/html; charset=utf-8'
-    meta = META_ROBOTS_RE.search(resp.content)
+    meta = META_ROBOTS_RE.search(resp.text)
     assert meta
     content = meta.group('content')
     if is_indexed:
@@ -248,7 +248,7 @@ LOCALE_SELECTORS = {
 @pytest.mark.nondestructive
 @pytest.mark.parametrize('expected,accept,cookie,param',
                          LOCALE_SELECTORS.values(),
-                         ids=LOCALE_SELECTORS.keys())
+                         ids=list(LOCALE_SELECTORS))
 @pytest.mark.parametrize(
     'slug', ['/search',
              '/events',

--- a/tests/headless/test_robots.py
+++ b/tests/headless/test_robots.py
@@ -17,9 +17,9 @@ def test_robots(any_host_url):
     urlbits = urlsplit(any_host_url)
     hostname = urlbits.netloc
     if hostname in INDEXED_ATTACHMENT_DOMAINS:
-        assert response.content.strip() == ''
+        assert response.text.strip() == ''
     elif hostname in INDEXED_WEB_DOMAINS:
-        assert 'Sitemap: ' in response.content
-        assert 'Disallow: /admin/\n' in response.content
+        assert 'Sitemap: ' in response.text
+        assert 'Disallow: /admin/\n' in response.text
     else:
-        assert 'Disallow: /\n' in response.content
+        assert 'Disallow: /\n' in response.text

--- a/tests/utils/urls.py
+++ b/tests/utils/urls.py
@@ -122,12 +122,16 @@ def assert_valid_url(url, location=None, status_code=requests.codes.moved_perman
             # strip off query for further comparison
             resp_location = resp_location.split('?')[0]
 
-        assert location == unquote(resp_location).decode()
+        assert location == unquote(resp_location)
 
     if resp_headers and not follow_redirects:
+
+        def convert_to_set(header):
+            return frozenset(d.strip() for d in header.lower().split(','))
+
         for name, value in resp_headers.items():
             assert name in resp.headers
-            assert resp.headers[name].lower() == value.lower()
+            assert convert_to_set(resp.headers[name]) == convert_to_set(value)
 
 
 # https://github.com/mozilla/bedrock/blob/master/tests/redirects/base.py

